### DISCUSSION TITLE
5.0.0beta1

### DISCUSSION THREAD

**Discussion by ktgw0316** [Aries85/LightZone#292] [Announcements]
Windows packages are made with [install4j](https://www.ej-technologies.com/products/install4j/overview.html), multi-platform installer builder.

To install LightZone on Linux, follow [wiki page](https://github.com/ktgw0316/LightZone/wiki#linux).

# Changes & Improvements

- Show thumnail/preview of Canon .cr3 raw files
  - Note: This version **cannot open/edit** the .cr3 files, since migration to LibRaw (#270) is still ongoing.
- Modern Look & Feel with [FlatLaf](https://www.formdev.com/flatlaf/) (#273)

# Bug Fixes

- Faint white grid lines over exported images (#162, #288)
- Cannot build with JDK19/20 (related: #287)

# Known Issue

- Cannot open Canon .cr3 raw files (#263)
- Missing most metadata from Canon .cr3 raw files
- Italian Help file is missing on Windows (#291)
- The 'lack of tile' artifact in Editor mode still sometimes occurs on macOS 

# Need tests

- If you cannot open your raw file that can be opened with older version of LightZone, please let me (@ktgw0316) know via issue or discussion page.

<hr /><em>This discussion was created from the release <a href='https://github.com/ktgw0316/LightZone/releases/tag/5.0.0beta1'>5.0.0beta1</a>.</em>

**Comment by ktgw0316** [comment]
Can't open Fujifilm raws. Maker name should be "FUJIFILM", not "FUJI" as in `com/lightcrafts/utils/DCRaw.java`.

**Reply by sfink16** [reply]
I have a weird one where a JPG is not recognized yet opens up in file viewer.  See the screenshot attached.  The JPG apparently is too big to attach.
![Screenshot from 2023-07-01 19-16-05](https://github.com/ktgw0316/LightZone/assets/62555600/833da0aa-7ae0-47b6-8fa4-f06139f2fac5)

**Comment by sfink16** [comment]
I cannot open Canon 5D MKIV CR2 images that I thought I could open in the previous version.

Ubuntu 22.04

**Reply by MJacal** [reply]
I can confirm this, using some samples from DPReview, that versions 4.1.9 and 4.2.4 can open, with tone curve included.

**Comment by ghost** [comment]
Ubuntu Focal Fossa

I cannot open .NEF raw files taken with a Nikon Z7. The error messages says "Unsupported Camera".
Raw settings are NEF compression on, color depth 14 bit.

**Reply by sfink16** [reply]
Which release are you using?  Has it opened in earlier release of LZ, such as 4.2.4.  I looked up the camera and it came out in late 2018.  The convertor that LZ uses is DCRaw, which ended it's production around that time.  LZ may not be able to handle the Z7 until it upgrade to LibRaw, something that the developer is working on.  See [#270](https://github.com/ktgw0316/LightZone/issues/270)?

Steve

**Reply by ghost** [reply]
Problem is in Lightzone 5.0.0.beta1, hence the reason to write it here.
I don't know if it would have worked in older Lightzone versions as this was my first try with photos from a Z7.

**Comment by sfink16** [comment]
> Problem is in Lightzone 5.0.0.beta1, hence the reason to write it here. I don't know if it would have worked in older Lightzone versions as this was my first try with photos from a Z7.

I asked about previous versions because of this NOTE on this page "If you cannot open your raw file that can be opened with older version of LightZone, please let me (@ktgw0316) know via issue or discussion page." The developer was interested what was working previously and broken now.

**Comment by doobes** [comment]
Raw files from my Fuji X100f are reported as unrecognized.

Also all references to www.lightzoneproject.org should prolly be removed.

Help -> Learning Center
Help -> Lightzone Project Page

Help -> About LightZone

Also, the help screen is not closed when exiting LZ.

**Comment by sfink16** [comment]
> Also all references to [www.lightzoneproject.org](http://www.lightzoneproject.org) should prolly be removed.
> 
> Help -> Learning Center Help -> Lightzone Project Page
> 
> Help -> About LightZone
> 
> Also, the help screen is not closed when exiting LZ.

Agreed the obsolete lightzone.project.org references should be removed.  That said, my help screen does close when using Ubuntu 22.04.  When clicking on "About Lightzone", I have to click on it again to close it before proceeding with further actions, including exiting the program.  Which OS are you using where the Help screen is not closing?

**Reply by doobes** [reply]
Sorry Windows 10.

I'm running the beta in a Microsoft Sandbox.

Cheers

**Reply by sfink16** [reply]
> Sorry Windows 10.
> 
> I'm running the beta in a Microsoft Sandbox.
> 

Interesting, beta is not working at all on windows home edition.

**Comment by Giant2000** [comment]
I can't open and edit NEF files anymore, RAW files made by Nikon D800. Camera was supported in 4.2.4 but not in 4.2.5 nor the beta 5.0.0
Get a "Could not initialize class com.lightcrafts.utils.raw.DCRaw" note. JPG workes fine.

**Reply by sfink16** [reply]
> I can't open and edit NEF files anymore, RAW files made by Nikon D800. Camera was supported in 4.2.4 but not in 4.2.5 nor the beta 5.0.0
> Get a "Could not initialize class com.lightcrafts.utils.raw.DCRaw" note. JPG workes fine.

Review the EXIF data in those files to see if the name matches exactly in this supported camera list in this document -> https://github.com/ktgw0316/LightZone/blob/master/lightcrafts/src/main/java/com/lightcrafts/utils/raw/DCRaw.java

**Comment by Marco1060** [comment]
Version 5.0.0 beta2 don't work with pentax file PEF
OS Ubuntu 24.04
Camera Pentax Kx
With older version on Ubuntu 22.04 no problem
![Schermata del 2024-12-08 10-59-55](https://github.com/user-attachments/assets/f6c4cd69-0f73-4979-bb47-d3c929c2cf21)

**Comment by HinterWelt** [comment]
Version 5.0.0 beta2 don't work with Nikon D500 RAW.nef (also my Z 7II)
OS Ubuntu 24.04
Camera Nikon D500
Worked with older version of UIbuntu same as above
![image](https://github.com/user-attachments/assets/7c8b1e6a-8d26-4ea6-aca0-88102dd32180)

**Reply by sfink16** [reply]
> Version 5.0.0 beta2 don't work with Nikon D500 RAW.nef (also my Z 7II)
> OS Ubuntu 24.04
> Camera Nikon D500

Review the EXIF data in those files to see if the name matches exactly in this supported camera list in this document -> https://github.com/ktgw0316/LightZone/blob/master/lightcrafts/src/main/java/com/lightcrafts/utils/raw/DCRaw.java

**Comment by HinterWelt** [comment]
NIKON D500 = [Github Found](https://github.com/ktgw0316/LightZone/blob/a502a67f78efefe53885bcefbfda43d5de008416/lightcrafts/src/main/java/com/lightcrafts/utils/raw/DCRaw.java#L1210)

NIKON Z 6_2 = Not found

**Reply by sfink16** [reply]
> NIKON Z 6_2 = Not found

LZ uses DCRaw for older cameras yet DCRaw has not updated in over 5 years.  For newer cameras LZ uses LibRaw.  Here is a supported camera list for Libraw -> https://www.libraw.org/supported-cameras-202403
It looks like your camera is on the list but has not been incorporated into LZ as of yet.  Remember, LZ has only one developer and he lives in Japan.  One thing you can try is downloading Libraw and running it.  Libraw is primitive, even more-so than LZ, but it does work. [Download](https://www.libraw.org/download
NOTE: use the Install instructions file to install on your Ubuntu.
.

**Reply by Marco1060** [reply]
thanks
i tried to do what you told me but it doesn't work i will try again when i have more time, in the meantime i will use another software, i am trying darktable.
P.S
I apologize for my bad english but i am an old italian and i am not very familiar with the language

**Reply by sfink16** [reply]
> i tried to do what you told me but it doesn't work i will try again when i have more time, in the meantime i will use another software, i am trying darktable.

Darktable is a good software, but I am starting to really enjoy DigiKam.  DigiKam also works great in culling images and is a natural album/folder organizer.  Both can save in 16-bit Tiffs to allow you to continue to use LZ if you still want to.  Both use Libraw to convert raw images.

**Reply by Marco1060** [reply]
Ok thanks for the info
